lib/avro/builder/dsl.rb

```diff
@@ -89,6 +89,25 @@ def type(*)
         @last_object = super
       end
 
+      # Several variants:
+      # define_type(:foo, long)
+      # define_type(:bar, { namespace: 'com.example' }, long)
```

Comment:
I played around with the different variants before arriving at this. I didn't like the name and namespace arguments being separated by the type object between which, which is why I ended up with this.

The alias will also inherit the namespace from the context in which its called, so I expect the actual usage to be more like:

```ruby
namespace 'com.example'

type_alias :bar, long
```

So we could possibly drop support for a namespace argument altogether. What do you think?